ci/run_python_tests.sh

```diff
@@ -0,0 +1,121 @@
+  deb_sha256_url=$(jq -r '.url' <<< "$deb_sha256_data")
+
+  deb_name=$(jq -r '.name' <<< "$deb_data")
+  deb_asc_name=$(jq -r '.name' <<< "$deb_asc_data")
+  deb_sha256_name=$(jq -r '.name' <<< "$deb_sha256_data")
+
+  [[ "$deb_name" == *"rc"* || "$deb_name" == *"bitcore"* ]] && is_valid=false
+  [[ "$deb_asc_name" == *"rc"* || "$deb_asc_name" == *"bitcore"* ]] && is_valid=false
+  [[ "$deb_sha256_name" == *"rc"* || "$deb_sha256_name" == *"bitcore"* ]] && is_valid=false
+
+  if [ "$is_valid" = true ]; then
+    break
+  fi
+done < <(jq -c '.[]' <<< "$json_data")
+
+curl -L ${deb_url} --output ${deb_name} || { val="$?"; echo "Error: was not able to download ${deb_name}."; exit $val; }
+curl -L ${deb_asc_url} --output ${deb_asc_name} || { val="$?"; echo "Error: was not able to download ${deb_asc_name}."; exit $val; }
+curl -L ${deb_sha256_url} --output ${deb_sha256_name} || { val="$?"; echo "Error: was not able to download ${deb_sha256_name}."; exit $val; }
+
+echo "" && echo "=== Checksum verification===" && echo ""
+if sha256sum -c "$deb_sha256_name"; then
+  echo "Checksum verification passed."
+else
+  echo "Checksum verification failed."
+  exit 1
+fi
+
+# Step 2
+echo "" && echo "=== Extract debian package content" && echo ""
+ZEN_CONTENT_FOLDER=${deb_name}-contents
+dpkg -x ${deb_name} ${ZEN_CONTENT_FOLDER} || { retval="$?"; echo "Error: was not able to extract package ${deb_name} to directory ${ZEN_CONTENT_FOLDER}."; exit $retval; }
+
+# Step 3
+echo "" && echo "=== Export BITCOINCLI, BITCOIND and SIDECHAIN_SDK path as env vars, needed for python tests" && echo ""
+BITCOINCLI=${CURRENT_DIR}/${ZEN_CONTENT_FOLDER}/usr/bin/zen-cli
+BITCOIND=${CURRENT_DIR}/${ZEN_CONTENT_FOLDER}/usr/bin/zend
+SIDECHAIN_SDK=${CURRENT_DIR}
+
+if [[ ! -f "$BITCOINCLI" ]]; then
+  fn_die "zen-cli does not exist in the given path. Exiting ..."
+fi
+if [[ ! -f "$BITCOIND" ]]; then
+  fn_die "zend does not exist in the given path. Exiting ..."
+fi
+if [[ ! -d "$SIDECHAIN_SDK" ]]; then
+  echo "Sidechain-SDK does not exist in the given path. Exiting ..."
+  exit 1
+fi
+
+export BITCOINCLI
+export BITCOIND
+export SIDECHAIN_SDK
+
+# Step 4
+echo "" && echo "=== Fetch zen params" && echo ""
+${CURRENT_DIR}/${ZEN_CONTENT_FOLDER}/usr/bin/zen-fetch-params || { retval="$?"; echo "Error: was not able to fetch zen params."; exit $retval; }
+
+# Step 5
+echo "" && echo "=== Building SideChain SDK ===" && echo ""
+mvn clean install -Dmaven.test.skip=true || { retval="$?"; echo "Error: was not able to complete mvn clean install of Sidechain SDK."; exit $retval; }
+
+# Step 6
+echo "" && echo "=== Installing node 16.0.0 ===" && echo ""
+source ~/.nvm/nvm.sh
+nvm install v16.0.0 || { retval="$?"; echo "Error: was not able to nvm install node 16.0.0"; exit $retval; }
+
+# Step 7
+echo "" && echo "=== Installing yarn ===" && echo ""
+npm install --global yarn || { retval="$?"; echo "Error: was not able to install yarn with npm install."; exit $retval; }
```

Comment:
Same for yarn https://docs.travis-ci.com/user/languages/javascript-with-nodejs/

Reply:
ditto

Reply:
@ysibirski 
Actually this is not needed, since node is installed from nvm, it makes sure that when installing `yarn` through npm it's always picking a version that is compatible with the node version you just installed

https://classic.yarnpkg.com/lang/en/docs/install/#mac-stable
https://askubuntu.com/questions/1020537/how-can-i-get-yarn-to-use-the-right-version-of-node

`It is recommended to install Yarn through the [npm package manager](http://npmjs.org/), which comes bundled with [Node.js](https://nodejs.org/) when you install it on your system.`